Automattic/wp-calypso
# Site Credit Block Structure

For now, I'm looking for feedback on if this looks like the right direction for the block. We'll continue using the site option to store the data on dotcom, but the main changes we'll need (for the future) are adding integration points to override settings in a dotcom context (i.e. hooks for whether or not the block is enabled, messages to display prompting for an upgrade, etc)

#### To do for this PR:
- [x] Allow users to align the block left/right/center
- [x] Use the footercredit option for dynamic rendering in PHP.
- [x] Update Varia to not render a footer credit if FSE is enabled. (i.e. relying on the block instead)
- [x] (possibly for followup) Refactor some of the `useSiteOptions` code so it doesn't need to have so many duplicated props.

#### Changes proposed in this Pull Request
* Create site credit block files and initial structure.
* Register `footercredit` as a setting so we can access it from the API. This is the same option name used to store the data currently on .com, so we'll be able to integrate easily with the existing solution.

Current functionality:
![2019-10-29 22 57 31](https://user-images.githubusercontent.com/6265975/67832702-b41c2c00-fa9f-11e9-98e5-7482442700fd.gif)

_Note: rendering based on the selection isn't working yet. For now, test the block editor code._

#### Testing instructions
* Pull this, insert the site credit block, and play around with it. Right now, the option is persisted correctly, but it is not rendered.
* Look at the code.
* If you insert the credit block into the footer and also use this theme PR: https://github.com/Automattic/themes/pull/1612, only the block version should show on the front end.

Resolves #37045.
